Lilu/Headers/kern_util.hpp

```diff
@@ -944,4 +944,206 @@ class evector_base {
 template <typename T, void (*deleter)(T)=emptyDeleter<T>>
 class evector : public evector_base<typename remove_reference<T>::type, T, deleter> { };
 
+/**
+ *  Represents a circular buffer protected by a recursive mutex lock
+ */
+template <typename T>
+struct CircularBuffer {
+private:
+	/**
+	 *  The internal storage
+	 */
+	T *storage;
```

Comment:
Please provide inline field initialisation. I.e. {} or so. For fields that will be overwritten later in init please still use zeroing, that's a safety convention.